### PR TITLE
Improve nixos

### PR DESCRIPTION
**Pull request by misumisumi**
#1211 

- Support for placing `lazy-config.json` under `.config/nvim`
- Change from `nixosModules` to `homeManagerModules` (BREAKING CHANGE)

### CODE REVIEW

**Comment by ayamir** [comment]
Thanks for your continuous contributions! I don't have `nix` or `nixos` environment so can't test it. Could you please check it? @aarnphm @CharlesChiuGit

File: nixos/neovim/default.nix [hunk 1]
```diff
@@ -89,32 +94,21 @@ in
...
-        name = "nvim-depends-include";
-        paths = splitString ":" (makeIncludePath build-dependent-pkgs);
-        extraPrefix = "/lib/nvim-depends/include";
-        ignoreCollisions = true;
-      };
-      nvim-depends-pkgconfig = pkgs.buildEnv {
-        name = "nvim-depends-pkgconfig";
-        paths = splitString ":" (makePkgConfigPath build-dependent-pkgs);
-        extraPrefix = "/lib/nvim-depends/pkgconfig";
+      neovim-build-deps = pkgs.buildEnv {
```
Comment (aarnphm) [inline nixos/neovim/default.nix:97]: qq: why do we have to build neovim from scratch again?

Can we utilize something like neovim-nightly or sth?

Comment (CharlesChiuGit) [reply nixos/neovim/default.nix:97]: i can add that, but i'm half way on completing my nix config.
i'll check this pr once i'm done.

Comment (aarnphm) [reply nixos/neovim/default.nix:97]: sg. I have yet to migrate all my neovim to nix yet so haven't tested it extensively.

**Comment by ayamir** [comment]
Does this PR work as expect?

File: nixos/neovim/default.nix (hunk 1 above)
Comment (CharlesChiuGit) [reply nixos/neovim/default.nix:100]: i guess i'm wrong, it's not the same as adding overlays in home manager's `home.nix`.

**Comment by CharlesChiuGit** [comment]
@misumisumi 
why do we need to include `dotnet`? is that necessary?

**Comment by misumisumi** [comment]
This is a little difficult when using dotnet in a NIX environment, so we made it modular to make it easier to configure.
Actually, there is a procedure in A, so there is no problem even if it is not included here.

**Pull request merged**